assets/js/SignPanel.tsx

```diff
@@ -203,19 +183,9 @@ function SignPanel({
     }
   }
 
-  function saveStaticText(): void {
+  function save(config: SignConfig): void {
     setCustomChanges(false);
-
-    const newConfig: SignConfig = {
-      ...signConfig,
-      mode: 'static_text',
-      line1: staticLine1,
-      line2: staticLine2,
-    };
-
-    setConfigs({
-      [realtimeId]: newConfig,
-    });
+    setConfigs({ [realtimeId]: config });
```

Comment:
Just as the sign is passed only its own config (not all of them), could we pass each sign an update function that only updates its own config? Then individual signs wouldn't have to "know" the data model of the larger config-for-all-signs.

Reply:
Pushed this fix up in a separate fixup commit because I'm kind of iffy on it now. It's certainly much cleaner, but now there's an explosion of memoized new functions. (In contrast to the individual config, where each `SignPanel` is just passed its own pointer into the same larger structure.) Still, these are sort of micro-optimizations so might be worth it for the improved clarity.